assets/common/src/register.py

```diff
@@ -141,14 +145,16 @@ def main():
 
     # validations
     if model_type not in SUPPORTED_MODEL_ASSET_TYPES:
-        raise AzureMLException._with_error(
-                AzureMLError.create(UnSupportedModelTypeError, model_type=model_type)
-            )
+        raise AzureMLException._with_error(AzureMLError.create(UnSupportedModelTypeError, model_type=model_type))
 
     if not model_name:
-        raise AzureMLException._with_error(
-            AzureMLError.create(MissingModelNameError)
-        )
+        raise AzureMLException._with_error(AzureMLError.create(MissingModelNameError))
+
+    if not re.match(VALID_MODEL_NAME_PATTERN, model_name):
+        # update model name to one supported for registration
+        logger.info(f"Updating model name to match pattern `{VALID_MODEL_NAME_PATTERN}`")
+        model_name = re.sub(NEGATIVE_MODEL_NAME_PATTERN, "-", model_name)
+        logger.info(f"Updated model_name = {model_name}")
 
```

Comment:
Would the customer know if they have registered the model with an incorrect naming convention and we changed it? and to what we've changed it?

Reply:
1. There is a log that I have added in the PR for name change.
2. registration asset id is logged for successful registration which would contain updated model name
3. registeration_details would contain updated model name

please let me know if we need to improve logging or share details by some other means